src/tools/build_helper/src/git.rs

```diff
@@ -78,22 +78,13 @@ pub fn rev_exists(rev: &str, git_dir: Option<&Path>) -> Result<bool, String> {
 /// We will then fall back to origin/master in the hope that at least this exists.
 pub fn updated_master_branch(git_dir: Option<&Path>) -> Result<String, String> {
     let upstream_remote = get_rust_lang_rust_remote(git_dir)?;
-    for upstream_master in [format!("{upstream_remote}/master"), format!("origin/master")] {
```

Comment:
Is it worth checking a revert of just the changes in updated_master_branch? Those are unrelated to the Miri changes, it just seemed reasonable to ensure the remove branch returned here actually exists.